lisa/sut_orchestrator/hyperv/context.py

```diff
@@ -1,21 +1,37 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from pathlib import PurePath
-from typing import Optional
+from typing import List, Optional
 
 from lisa import Node, RemoteNode
+from lisa.sut_orchestrator.hyperv.schema import DeviceAddressSchema
+from lisa.sut_orchestrator.util.schema import HostDevicePoolType
 from lisa.util.process import Process
 
 
+@dataclass
+class DevicePassthroughContext:
+    pool_type: HostDevicePoolType = HostDevicePoolType.PCI_NIC
+    device_list: List[DeviceAddressSchema] = field(
+        default_factory=list,
+    )
+
+
 @dataclass
 class NodeContext:
     vm_name: str = ""
     host: Optional[RemoteNode] = None
     working_path = PurePath()
     serial_log_process: Optional[Process] = None
 
+    # Device pass through configuration
+    is_device_passthrough_set: bool = False
+    device_passthrough_context: List[DevicePassthroughContext] = field(
```

Comment:
1. `passthrough_devices` is a better name.
2. Because the devices are mainly used by types, so the type `dict[HostDevicePoolType, List[DeviceAddressScehma]` is easier to use.

Reply:
> 1. `passthrough_devices` is a better name.
> 2. Because the devices are mainly used by types, so the type `dict[HostDevicePoolType, List[DeviceAddressScehma]` is easier to use.

@squirrelsc 

1> Sure, I will make the change for name as `passthrough_devices `

2> Idea is to set the list of devices for given node context and put only devices which will be passthrough. We dont need to track pool type here. Whatever is there in list, we will assign it to node. Pool type is needed while requesting device from the pool. Node context does not need the pool type.

Reply:
For #2, you need to go through the collection (either dict or list), and find the right one by pool_type, when using the context. So, dict is easier to query than list. If you don't need to know the pool type, and most operations are to iterate all items, list is better.